src/Nethermind/Nethermind.Network.Stats/NodeStatsManager.cs

```diff
@@ -49,10 +54,39 @@ public int GetHashCode(Node obj)
         
         private readonly ILogger _logger;
         private readonly ConcurrentDictionary<Node, INodeStats> _nodeStats = new ConcurrentDictionary<Node, INodeStats>(new NodeComparer());
+        private readonly ITimer _cleanupTimer;
+        private readonly int _maxCount;
 
-        public NodeStatsManager(ILogManager logManager)
+        public NodeStatsManager(ITimerFactory timerFactory, ILogManager logManager, int maxCount = 10000)
         {
+            _maxCount = maxCount;
             _logger = logManager?.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
+
+            _cleanupTimer = timerFactory.CreateTimer(TimeSpan.FromMinutes(10));
+            _cleanupTimer.Elapsed += CleanupTimerOnElapsed;
+            _cleanupTimer.Start();
+        }
+
+        private void CleanupTimerOnElapsed(object sender, EventArgs e)
+        {
+            int deleteCount = _nodeStats.Count - _maxCount;
+
+            if (deleteCount > 0)
+            {
+                IEnumerable<Node> toDelete = _nodeStats
```

Comment:
I think this will cause concurrency issues. These methods are not thread safe so I think you should lock _nodeStats here.